include/ck/tensor_operation/gpu/element/unary_element_wise_operation.hpp

```diff
@@ -0,0 +1,91 @@
+#pragma once
+#include "data_type.hpp"
+#include "math_v2.hpp"
+
+namespace ck {
+namespace tensor_operation {
+namespace element_wise {
+
+struct PassThrough
+{
+    template <typename T>
+    __host__ __device__ void operator()(T& y, const T& x) const
+    {
+        static_assert(is_same<T, float>::value || is_same<T, double>::value ||
+                          is_same<T, half_t>::value || is_same<T, bhalf_t>::value ||
+                          is_same<T, int32_t>::value || is_same<T, int8_t>::value,
+                      "Data type is not supported by this operation!");
+
+        y = x;
+    };
+};
+
+struct UnaryDivide
+{
+    __host__ __device__ UnaryDivide(const int32_t divider = 1) : divider_(divider){};
+
+    template <typename T>
+    __host__ __device__ void operator()(T& y, const T& x) const;
+
+    template <>
+    __host__ __device__ void operator()<float>(float& y, const float& x) const
+    {
+        y = x / type_convert<float>(divider_);
+    };
```

Comment:
Why not use static_assert?

Reply:
Updated by  [1a4ee](https://github.com/ROCmSoftwarePlatform/composable_kernel/pull/274/commits/61d01e71055f76b34b5a5e3b9f064f6cfbf1a4ee)

Reply:
We could use is_same and following implementation
y = x / type_convert<T>(divider_);

Reply:
Do you mean `y = x / type_convert<T>(divider_);` ?

Reply:
Updated by  [c8965](https://github.com/ROCmSoftwarePlatform/composable_kernel/pull/274/commits/aefaf5ee3a8e320b474533d6fe8c6448f32c8965)